UnityProject/Assets/Scripts/Objects/Doors/ConstructibleDoor.cs

```diff
@@ -89,7 +89,7 @@ public bool CheckBolts()
 			}
 			else
 			{
-				return boltsModule.CanDoorStateChange();
+				return boltsModule.BoltsDown!;
```

Comment:
What's the exclamation mark at the end?

Reply:
doesn't it invert?

Reply:
at the end?

Reply:
``return !boltsModule.BoltsDown``

Reply:
idk, what does it do at the end that's a question?

Reply:
just seems to do nothing need more investigation sometime, fixed the

Reply:
at the end of anything (or in the middle) it means "trust me bro, this isn't null".

so ``boltsModule.BoltsDown!`` means "trust me, ``BoltsDown`` isn't null".
while ``boltsModule!.BoltsDown`` means "trust me, my reference of ``boltsModule`` isn't null.

Reply:
I use it on typescript so the typescript *compiler* doesn't think something might be undefined and auto assign the type ``type | undefined``.